src/github_runner_image_builder/builder.py

```diff
@@ -0,0 +1,635 @@
+
+    Raises:
+        DependencyInstallError: If there was an error installing apt packages.
+    """
+    try:
+        output = subprocess.check_output(
+            ["/usr/bin/apt-get", "update", "-y"], encoding="utf-8", timeout=30 * 60
+        )  # nosec: B603
+        logger.info("apt-get update out: %s", output)
+        output = subprocess.check_output(
+            ["/usr/bin/apt-get", "install", "-y", "--no-install-recommends", *APT_DEPENDENCIES],
+            encoding="utf-8",
+            timeout=30 * 60,
+        )  # nosec: B603
+        logger.info("apt-get install out: %s", output)
+        output = subprocess.check_output(
+            ["/usr/bin/snap", "install", SNAP_GO, "--classic"],
+            encoding="utf-8",
+            timeout=30 * 60,
+        )  # nosec: B603
+        logger.info("apt-get snap install go out: %s", output)
+    except subprocess.CalledProcessError as exc:
+        logger.exception(
+            "Error installing dependencies, cmd: %s, code: %s, err: %s",
+            exc.cmd,
+            exc.returncode,
+            exc.output,
+        )
+        raise DependencyInstallError from exc
+
+
+def _enable_network_block_device() -> None:
+    """Enable network block device module to mount and build chrooted image.
+
+    Raises:
+        NetworkBlockDeviceError: If there was an error enable nbd kernel.
+    """
+    try:
+        output = subprocess.check_output(["/usr/sbin/modprobe", "nbd"], timeout=10)  # nosec: B603
+        logger.info("modprobe nbd out: %s", output)
+    except subprocess.CalledProcessError as exc:
+        logger.exception(
+            "Error enabling network block device, cmd: %s, code: %s, err: %s",
+            exc.cmd,
+            exc.returncode,
+            exc.output,
+        )
+        raise NetworkBlockDeviceError from exc
+
+
+def build_image(arch: Arch, base_image: BaseImage) -> None:
+    """Build and save the image locally.
+
+    Args:
+        arch: The CPU architecture to build the image for.
+        base_image: The ubuntu image to use as build base.
+
+    Raises:
+        BuildImageError: If there was an error building the image.
+    """
+    IMAGE_MOUNT_DIR.mkdir(parents=True, exist_ok=True)
+    try:
+        logger.info("Cleaning build state.")
+        _clean_build_state()
+        logger.info("Downloading base image.")
+        base_image_path = _download_base_image(arch=arch, base_image=base_image)
+        logger.info("Resizing base image.")
+        _resize_image(image_path=base_image_path)
+        logger.info("Mounting network block device.")
+        _mount_image_to_network_block_device(image_path=base_image_path)
+        logger.info("Replacing resolv.conf.")
+        _replace_mounted_resolv_conf()
+        logger.info("Resizing partitions.")
+        _resize_mount_partitions()
+        logger.info("Installing YQ from source.")
+        _install_yq()
+    except ImageBuilderBaseError as exc:
+        raise BuildImageError from exc
```

Comment:
I think `ImageBuilderBaseError` can inherit from `BuildImageError` which would avoid nesting the error

Note that due to the previous comment, the except might be needed to allow a finally, so you'd just have to raise